framework/auth/core.py

```diff
@@ -837,6 +842,23 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def confirm_token(self, token):
+        """Return whether or not a confirmation token is valid for this user.
+        :rtype: bool
+        """
+        try:
+            verification = self.email_verifications[token]
+        except:
+            return False
+        # Check token for existance and date
+        if (
+            'expiration' in verification and
+            verification['expiration'] < dt.datetime.utcnow() or
+            token not in self.email_verifications
+        ):
```

Comment:
Can you explicitly group the parts of this conditional for evaluation order? `x and (y or z)` or `(x and y) or z`